pyspectrafuse/commands/spectrum2msp.py

```
@@ -81,130 +86,333 @@ def create_consensus_strategy(method_type: str, sim: str = 'dot',
+            SELECT
+                {run_col} AS run_file_name,
+                {scan_expr} AS orig_scan,
+                mz_array,
+                intensity_array,
+                CAST({charge_col} AS INTEGER) AS charge,
+                {mz_expr} AS pepmass,
+                {pep_col} AS posterior_error_probability,
+                {qvalue_expr} AS global_qvalue
+            FROM read_parquet('{ppath}')
+            WHERE {charge_col} = {charge_int}
+        """)
```

Ruff (0.15.10)

[error] Possible SQL injection vector through string-based query construction

(S608)